packages/sdk/typescript/human-protocol-sdk/src/escrow.ts

```diff
@@ -483,40 +513,68 @@ export class EscrowClient extends BaseEthersClient {
    * await escrowClient.storeResults('0x62dD51230A30401C455c8398d06F85e4EaB6309f', 'http://localhost/results.json', 'b5dad76bf6772c0f07fd5e048f6e75a5f86ee079');
    * ```
    */
+  async storeResults(
+    escrowAddress: string,
+    url: string,
+    hash: string,
+    txOptions?: Overrides
+  ): Promise<void>;
+
   @requiresSigner
   async storeResults(
     escrowAddress: string,
     url: string,
     hash: string,
-    txOptions: Overrides = {}
+    a?: bigint | Overrides,
+    b?: Overrides
   ): Promise<void> {
+    const escrowContract = this.getEscrowContract(escrowAddress);
+
+    const isBigInt = typeof a === 'bigint';
+    const fundsToReserve = isBigInt ? (a as bigint) : undefined;
+    const txOptions = (isBigInt ? b : a) || {};
+
     if (!ethers.isAddress(escrowAddress)) {
       throw ErrorInvalidEscrowAddressProvided;
     }
-
     if (!url) {
       throw ErrorUrlIsEmptyString;
     }
-
     if (!isValidUrl(url)) {
       throw ErrorInvalidUrl;
     }
-
     if (!hash) {
       throw ErrorHashIsEmptyString;
     }
-
     if (!(await this.escrowFactoryContract.hasEscrow(escrowAddress))) {
       throw ErrorEscrowAddressIsNotProvidedByFactory;
     }
 
     try {
-      const escrowContract = this.getEscrowContract(escrowAddress);
-
-      await (await escrowContract.storeResults(url, hash, txOptions)).wait();
-
-      return;
+      if (fundsToReserve !== undefined) {
+        await (
+          await escrowContract['storeResults(string,string,uint256)'](
+            url,
+            hash,
+            fundsToReserve,
+            txOptions
+          )
+        ).wait();
+      } else {
+        await (
+          await escrowContract['storeResults(string,string)'](
+            url,
+            hash,
+            txOptions
+          )
+        ).wait();
+      }
     } catch (e) {
+      if (!isBigInt && e.reason === 'Forbidden') {
+        throw ErrorStoreResultsVersion;
+      }
+      // eslint-disable-next-line no-console
+      console.warn(WarnVersionMismatch);
```

Comment:
Let's avoid writing console logs from sdk (at least w/o explicitly specifying it via some configuration params (e.g. `HP_SDK_DEBUG_LOG` env var; or `networkData.chainId == ChainId.LOCALHOST` check), otherwise it pollutes stdout of application where it runs and creates inconvenience in log management

Reply:
I use the warn to kind of show it as a hint. Then I throw the actual error. I dont want to modify the error object/message